include/operations/blas_operators.hpp

```diff
@@ -213,8 +221,34 @@ SYCLBLAS_DEFINE_BINARY_OPERATOR(minOp2_struct, const_val::max,
                                 ((l < r) ? l : r))
 SYCLBLAS_DEFINE_BINARY_OPERATOR(addAbsOp2_struct, const_val::zero,
                                 (syclblas_abs::eval(l) + syclblas_abs::eval(r)))
-SYCLBLAS_DEFINE_TERNARY_OPERATOR(maxIndOp3_struct, const_val::min, > )
-SYCLBLAS_DEFINE_TERNARY_OPERATOR(minIndOp3_struct, const_val::max, < )
+
+struct maxIndOp2_struct {
```

Comment:
Instead of creating new struct can't you use the existing one through creating your own macro here